### PR TITLE
Added better comment recognition

### PR DESCRIPTION
**Pull request by zsmoore**
#102 #123 

# Description

This PR cleans up the interpreter by changing how we handle comments.
New strategy is to regex replace out any comments from the code we are interpreting.

**Note:** This PR breaks line highlighting but will be fixed in one of the upcoming PR's that changes how we think about syntax checking

Added regex checking on variable names to be safer.

# Change Log

-`interpret.js` : Code cleaned

# Manual Test Steps
Boot up the app and test that comments do not break anything.
commentfix

### CODE REVIEW

**Comment by zsmoore** [comment]
Added commit to address #123

File: src/interpreter/executor/interpret.js [hunk 1]
```diff
@@ -429,36 +387,7 @@ function masterRepToString(representation) {
...
...99-\u0fbc\u0fc6\u102b-\u103e\u1040-\u1049\u1056-\u1059\u105e-\u1060\u1062-\u1064\u1067-\u106d\u1071-\u1074\u1082-\u108d\u108f-\u109d\u135d-\u135f\u1712-\u1714\u1732-\u1734\u1752\u1753\u1772\u1773\u17b4-\u17d3\u17dd\u17e0-\u17e9\u180b-\u180d\u1810-\u1819\u18a9\u1920-\u192b\u1930-\u193b\u1946-\u194f\u19b0-\u19c0\u19c8\u19c9\u19d0-\u19d9\u1a17-\u1a1b\u1a55-\u1a5e\u1a60-\u1a7c\u1a7f-\u1a89\u1a90-\u1a99\u1b00-\u1b04\u1b34-\u1b44\u1b50-\u1b59\u1b6b-\u1b73\u1b80-\u1b82\u1ba1-\u1bad\u1bb0-\u1bb9\u1be6-\u1bf3\u1c24-\u1c37\u1c40-\u1c49\u1c50-\u1c59\u1cd0-\u1cd2\u1cd4-\u1ce8\u1ced\u1cf2-\u1cf4\u1dc0-\u1de6\u1dfc-\u1dff\u200c\u200d\u203f\u2040\u2054\u20d0-\u20dc\u20e1\u20e5-\u20f0\u2cef-\u2cf1\u2d7f\u2de0-\u2dff\u302a-\u302f\u3099\u309a\ua620-\ua629\ua66f\ua674-\ua67d\ua69f\ua6f0\ua6f1\ua802\ua806\ua80b\ua823-\ua827\ua880\ua881\ua8b4-\ua8c4\ua8d0-\ua8d9\ua8e0-\ua8f1\ua900-\ua909\ua926-\ua92d\ua947-\ua953\ua980-\ua983\ua9b3-\ua9c0\ua9d0-\ua9d9\uaa29-\uaa36\uaa43\uaa4c\uaa4d\uaa50-\uaa59\uaa7b\uaab0\uaab2-\uaab4\uaab7\uaab8\uaabe\uaabf\uaac1\uaaeb-\uaaef\uaaf5\uaaf6\uabe3-\uabea\uabec\uabed\uabf0-\uabf9\ufb1e\ufe00-\ufe0f\ufe20-\ufe26\ufe33\ufe34\ufe4d-\ufe4f\uff10-\uff19\uff3f]*$/.test(buffer);
```
Comment (angushtlam) [inline src/interpreter/executor/interpret.js]: omg

**Comment by angushtlam** [comment]
screen shot 2017-10-24 at 10 40 12 am

The line highlight doesn't match up with the actual line executed and shown in the visualizer

**Comment by zsmoore** [comment]
@angushtlam Take a look at the note I put in the PR, line highlighting will be broken until my next PR, I can add it into this one if you would like however it is something I am aware of and will be fixing.

**Comment by angushtlam** [comment]
@zsmoore Can you branch off of this branch for the fixes and make a PR from that branch into `interpreter`?

**Comment by zsmoore** [comment]
Going to drop this PR and open one for my new branch with an interpreter overhaul.

**Pull request closed without merging**